docs/reference/remap/functions/redact.cue

```diff
@@ -0,0 +1,83 @@
+package metadata
+
+remap: functions: redact: {
+	category:    "String"
+	description: """
+		Redact sensitive data in `value` such as:
+
+		- [US social security card numbers](\(urls.us_social_security_number))
+		- and other forms of personally identifiable information via custom patterns
+		- (more to come!)
+
+		This can help achieve compliance by ensuring sensitive data never leaves your network.
+		"""
+
+	arguments: [
+		{
+			name: "value"
+			description: #"""
+				The value to redact sensitive data from.
+
+				Its behavior differs depending on the type of `value`:
+
+				- For strings, it simply redacts the sensitive data and returns a new string
+				- For arrays, it redacts the sensitive data in each string element
+				- For objects, it masks the sensitive data in each string value, but not keys
+
+				For arrays and objects it will recurse into any nested arrays or objects. Any non-string elements will
+				be skipped.
+
+				Any redacted text will be replaced with `[REDACTED]`.
```

Comment:
Can we open a follow up issue to make this value configurable?

Reply:
👍 https://github.com/timberio/vector/issues/7445